department-of-veterans-affairs/abd-vro
# Update scripts and GH actions for SecRel 4.1

## What was the problem?


Our image-publishing scripts and GH actions relies on the target environment to determine which image prefix to use. 
Why? Some target environments (i.e., `sandbox`, `prod-test`, `prod`) require signed images. In SecRel 3, images were only signed for GH releases.

In SecRel 4, all images are signed (once they pass SecRel scans). This opens some avenues for us to deploy any image to any environment. However, our current scripts don't support this.

- [Slack](https://dsva.slack.com/archives/C01Q7979Z7D/p1676557786644419?thread_ts=1676503886.646189&cid=C01Q7979Z7D)
- #888

## How does this fix it?[^1]


Update scripts and GH actions that publish images to use an image prefix argument, instead of a target environment argument.

## How to test this PR
Manually run modified GH actions.
- ran [Deploy to Dev](https://github.com/department-of-veterans-affairs/abd-vro/actions/runs/4196235080), which calls [deploy-unsigned.yml](https://github.com/department-of-veterans-affairs/abd-vro/pull/1123/files#diff-0e445cdb471071791a52f5630a62c6ea2380921c1e2a0cef99d84097af1b6009)
- ran [SecRel 4](https://github.com/department-of-veterans-affairs/abd-vro-internal/actions/runs/4196214025)

Once PR is merged, disable SecRel 3 workflow and create a GH release, which should trigger SecRel 4 to run and publish images without the `dev_` prefix, which can then be used for `prod-test` and `prod` deployments.


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
